dvc/stage/loader.py

```diff
@@ -140,6 +145,8 @@ def __getitem__(self, name):
         except EntryNotFound:
             raise StageNotFound(self.dvcfile, name)  # noqa: B904
 
+        validate(resolved_data, Schema(SCHEMA))
```

Comment:
@skshetry Before I clean up the rest of the PR or tests, is this what you were thinking? We would still need to catch the errors on the first round of validation to get to this point, but at least we then use the existing schema to revalidate the resolved schema.

Reply:
@skshetry ping. WDYT about this approach?